src/components/editable-json.jsx

```diff
@@ -157,7 +157,9 @@ class JsonEditor extends React.Component {
    * @param {String} message - The error message.
    */
   handleUpdateError() {
-    this.setState({ mode: ERROR, message: this.props.updateError });
+    setTimeout(() => {
+      this.setState({ mode: ERROR, message: this.props.updateError });
+    }, 100);
```

Comment:
For these … this matches the other update result handlers, and as far as I can tell, the previous code broke the UI by creating an infinite `handleUpdateError() → setState() → componentDidUpdate() → handleUpdateError() → ...` loop … I would assume that this still isn’t ideal, but I also don’t know the ideal way to fix this (and, yeah, we also delay this in the other update handlers).

Reply:
Instead of calling `setTimeout` here, could we do an equality check on these items we're setting in the `state` and only call `setState` when they're in need of updating? That way it can only happen once.
I think this having a set timeout here might put us into the same loop, just throttled by 100ms.

It does look like the other methods here might get into the same loop as well.
Probably it would be best for us to have these state items on the component in a redux store which we can then update outside of any rendering cycle. I think that's way more work than we want to take on here though.

Reply:
Also moving things onto the store would allow us to keep things like expanded state when we unmount which would let us not have to keep everything rendered in https://github.com/mongodb-js/compass-collection/pull/135 which might be a big performance hit.

Made a jira ticket for it: https://jira.mongodb.org/browse/COMPASS-4485

Reply:
@Anemy Yes, switched this to an equality check :slightly_smiling_face: